buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -706,6 +706,66 @@ public static void copy(AsciiString src, int srcIdx, ByteBuf dst, int length) {
     }
 
     /**
+     * Creates a retained duplicate of the specified {@link ByteBuf}.
+     */
+    public static ByteBuf rduplicate(ByteBuf buf) {
+        final int readerIndex = buf.readerIndex();
+        final int writerIndex = buf.writerIndex();
+
+        final AbstractByteBuf duplicate;
+        if (buf instanceof SlicedAbstractByteBuf) {
+            final SlicedAbstractByteBuf oldSlice = (SlicedAbstractByteBuf) buf;
+            duplicate = PooledSlicedByteBuf.newInstance(oldSlice.unwrap(), 0, buf.capacity(), oldSlice.adjustment());
```

Comment:
IIRC the original intention in https://github.com/netty/netty/pull/4421 was to only pool "derived" buffers if the pooled allocator was used. The default behavior in the unpooled case would just be `operation().retain()`. Using the unpooled allocator is sometime used in conditions where references counting may be ignored (or not implemented completely/consistently) to simplify the user interaction pattern ... introducing pooling code in this code path could be problematic.